MekHQ/src/mekhq/campaign/log/AwardLogger.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright (C) 2018 MegaMek team
+ *
+ * This file is part of MekHQ.
+ *
+ * MekHQ is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * MekHQ is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with MekHQ.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package mekhq.campaign.log;
+
+import megamek.common.util.EncodeControl;
+import mekhq.campaign.personnel.Award;
+import mekhq.campaign.personnel.Person;
+
+import java.text.MessageFormat;
+import java.util.Date;
+import java.util.ResourceBundle;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * This class is responsible to control the logging of Award Log Entries.
+ * @author Miguel Azevedo
+ */
+public class AwardLogger {
```

Comment:
This is effectively a stateless class, and I see no reason to create an instance. Every method can be made static. Resource bundles are caches, so the lookup carries very little overhead and I don't see much advantage to saving that as a class variable, but if you think that's a concern it could be made a static field initialized lazily.